### ISSUE TITLE
Getting "Invalid database" error when trying to configure repo db on Azure SQL DB

### ISSUE THREAD

**Issue by AlreadyPicked** [trimble-oss/dba-dash#998]
I created a database on Azure SQL DB Server for the repo of DBA Dash but I can't seem to connect to it. Since Integrated security is not available, so used a SQL login/user.

After creating the login, I created the user in the repo and gave it the db_owner role.

Then I get this error :

![image](https://github.com/user-attachments/assets/1bfb2918-8f5e-4287-916d-023741ed680a)

If I add the user in the master database, I get this error instead :

![image](https://github.com/user-attachments/assets/576729a0-7c94-45ac-83ba-70db6e16dd55)

To me, it seems as it's not connecting to the user db and trying in master instead. Even if the connection string says : Initial Catalog=DBADash;

**Comment by DavidWiseman** [comment]
Hi,

There is a validation check to see if the database exists.  The initial catalog is removed from the connection string which means it will connect to the user default database (typically master).  This is done as the connection to the database will fail if the database doesn't exist. If the user has access to master now this should be OK.

The invalid database message occurs if the initial catalog is set to a database that has user objects that doesn't look like the DBA Dash repository database.  This is added as a safety check so the DBA Dash schema isn't applied to some user database by accident.

The check runs this query:

```
IF EXISTS(
	SELECT * FROM sys.extended_properties
	WHERE major_id=OBJECT_ID('DBVersionHistory')
	AND name = 'AppID'
	AND value='74F9FD8A-DF22-4355-9A7A-6E1F4AE712B9'
)
BEGIN 
	SELECT TOP(1) Version
	FROM dbo.DBVersionHistory
	ORDER BY DeployDate DESC
END
ELSE IF NOT EXISTS(SELECT * 
			FROM sys.objects
			WHERE type NOT IN('S','IT','SQ')          
            )
        AND DB_ID()>4
BEGIN
	SELECT '0.0.0.0' AS Version
END
ELSE
BEGIN 
	RAISERROR('Invalid database',11,1)
END
```

If the database is empty, check if this query returns any rows:

```
SELECT * 
FROM sys.objects
WHERE type NOT IN('S','IT','SQ')          
```

You would need to drop any objects returned in the list above. I typically allow the service to create the repository database rather than pre-creating the database.  This might be an option you could use instead and adjust the sizing of the DB afterward.

Hope this helps.

**Comment by AlreadyPicked** [comment]
Hi David,

I tested the query above in my DBADash database and it returns "Invalid database", even if the database is empty.

The first IF returns no rows. The ELSE IF return at least one row : 
![image](https://github.com/user-attachments/assets/bd790ac0-3b41-4731-95ba-c62f84f1c670)

So it defaults to the raiserror.

**Comment by DavidWiseman** [comment]
Hi thanks for reporting back - the issue will be fixed in the next release (soon).

As a workaround you can run this in the empty repository database.  It will allow it to detect the database as a DBA Dash repository database - allowing the schema to be applied.

```
CREATE TABLE [dbo].[DBVersionHistory](
	[DeployDate] [datetime2](3) NOT NULL,
	[Version] [varchar](20) NOT NULL,
 CONSTRAINT [PK_DBVersionHistory] PRIMARY KEY CLUSTERED 
(
	[DeployDate] ASC,
	[Version] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
) ON [PRIMARY]
GO

EXEC sys.sp_addextendedproperty @name=N'AppID', @value=N'74F9FD8A-DF22-4355-9A7A-6E1F4AE712B9' , @level0type=N'SCHEMA',@level0name=N'dbo', @level1type=N'TABLE',@level1name=N'DBVersionHistory'
GO
```

**Comment by DavidWiseman** [comment]
Fixed in 3.10 🚀
